### DISCUSSION TITLE
Insert entire list without duplication & UNIQUE constraint failed

### DISCUSSION THREAD

**Discussion by RzTutul** [vitusortner/floor#718] [Q&A] [answered]
**Here I want to insert some static categories when the app will be lunch. the first time it was inserted successfully but the second when the app lunch it show "UNIQUE constraint failed". I want if categories name already exists then it should be ignored **
here is my code :

@database(version: 1, entities: [CategoriesEntity])
abstract class AppDatabase extends FloorDatabase {
HishabDao get hishabDao;
}

**Entity**
https://github.com/entity(tableName : "cate_tbl",indices : [Index(value : ["cata_name"], unique : true)])
class CategoriesEntity {
@PrimaryKey(autoGenerate: true)
final int? id;
final String cata_name;
CategoriesEntity(this.id,this.cata_name);
}

**DAO**
@insert(onConflict : OnConflictStrategy.ignore)
Future insertCategories(List cate);

**List**
static final List listCate = [
CategoriesEntity(null,"Short"),
CategoriesEntity(null, "Medium"),
CategoriesEntity(null,"Large"),
];

void main() async{
WidgetsFlutterBinding.ensureInitialized();
final database = await $FloorAppDatabase.databaseBuilder('app_database.db').build();
final hishabDao = database.hishabDao;
await hishabDao.insertCategories(HelperUtils.listCate);
runApp(const MyApp());
}

**ERROR**

E/flutter (10713): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: DatabaseException(UNIQUE constraint failed: cate_tbl.cata_name (code 2067 SQLITE_CONSTRAINT_UNIQUE)) sql 'INSERT OR ABORT INTO cate_tbl (id, cata_name) VALUES (NULL, ?)' args [Small]

**Comment by dkaera** [comment, marked as answer]
the issue has been solved here #716
